### DISCUSSION TITLE
Register-PnPAzureADApp in Azure Cloud Shell CNG Issue

### DISCUSSION THREAD

**Discussion by anthonywhite** [pnp/powershell#381] [Q&A] [answered]
Would like to use Azure Cloud Shell for all things PnP.

Installed PnP.PowerShell 1.3.0 today

tried to use `Register-PnPAzureADApp` but got error:

`Register-PnPAzureADApp: Windows Cryptography Next Generation (CNG) is not supported on this platform.`

Is cloud shell a viable platform for PnP?

**Comment by erwinvanhunen** [comment, marked as answer]
I just checked and notice it too. We'll have to look into a solution for this. For now I would propose to register the azure ad app from a 'normal' PowerShell session, once registered you can use it from your Azure Cloud Shell. Other cmdlets should work normally in the Azure Cloud Shell.

**Reply by erwinvanhunen** [reply]
the error message occurs when we generate the self signed certificate. You could alternatively create your own certificate and use that: `Register-AzureADApp -CertificatePath <yourcert.pfx> [-CertificatePassword <password>]`

**Reply by anthonywhite** [reply]
Thanks Erwin, yes I worked around by using Powershell Core on Windows but useful to know about the other option 👍🏻
